### PR TITLE
Upgrade cert-manager to v1.12.8

### LINKED ISSUE #1910: Upgrade cert-manager to v1.13.2
**What Chart should be upgraded?**:

upstream / cert-manager https://github.com/cert-manager/cert-manager/releases/tag/v1.13.2
Please upgrade from 11 to 12 and then from 12 to 13
Please carefully read the release notes for [v1.12.0](https://github.com/cert-manager/cert-manager/releases/tag/v1.12.0) and [v1.13.0](https://github.com/cert-manager/cert-manager/releases/tag/v1.13.0) as there are potential breaking changes/

**Why should this Chart be upgraded?**:

- [ ] We need to perform a security upgrade.
- [ ] We need it for a new feature: *which feature (link to blocked issue)*.
- [x] We need to keep up to prevent future issues.
- [ ] Other: *Write the reason here*

**Acceptance criteria**:

- I checked the migration of the new Chart:
  - [ ] I upgraded a Chart and determined that no migration steps are needed.
  - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).
- [ ] I tested the functionality provided by the new Chart (e.g., Auth flow, Grafana dashboards, etc.)

### PR DESCRIPTION
**Pull request by lunkan93**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [ ] kind/admin-change  
- [x] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### Application Developer notice
Cert-manager Issuers and ClusterIssuers have been updated to use the [recommended](https://cert-manager.io/docs/configuration/acme/http01/#ingressclassname) `ingressClassName` field over the `class` field. We recommend that Application Developers use the `ingressClassName` field as well.

### What does this PR do / why do we need this PR?

Upgrades `cert-manager` from `v1.11.0` to `v1.12.8`

- Fixes #1910 

#### Additional information to reviewers
- Upgraded the chart
- Added migration script to move `class` to `ingressClassName` for issuers per recommendation from [upstream](https://cert-manager.io/docs/releases/release-notes/release-notes-1.12/#support-for-ingressclassname-in-the-http-01-solver)
- Exposed `extraArgs` for `cainjector` for if we ever would like to try things like `--enable-certificates-data-source=true` (original [PR](https://github.com/cert-manager/cert-manager/pull/5746) and [renaming](https://github.com/cert-manager/cert-manager/pull/5766) for context) for certain environments.
- Tested deploying the [user-demo](https://elastisys.io/compliantkubernetes/user-guide/deploy/#deploy-user-demo) - Have not run into any issues with Certificate creation.

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [ ] The change is transparent
  - [ ] The change is disruptive
  - [ ] The change requires no migration steps
  - [x] The change requires migration steps
- Metrics checks:
  - [ ] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [x] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [x] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [ ] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Comment by OlleLarsson** [comment]
> Added migration script to move class to ingressClassName for issuers per recommendation from [upstream](https://cert-manager.io/docs/releases/release-notes/release-notes-1.12/#support-for-ingressclassname-in-the-http-01-solver) 

This feels like something that we should advice applications developers to use as well

File: migration/v0.37/prepare/60-move-issuer-ingress-class.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+HERE="$(dirname "$(readlink -f "${0}")")"
+ROOT="$(readlink -f "${HERE}/../../../")"
+
+# shellcheck source=scripts/migration/lib.sh
+source "${ROOT}/scripts/migration/lib.sh"
+
+log_info "moving issuer ingress class keys"
+yq_move common .issuers.letsencrypt.prod.solvers[].http01.ingress.class .issuers.letsencrypt.prod.solvers[].http01.ingress.ingressClassName
+yq_move common .issuers.letsencrypt.staging.solvers[].http01.ingress.class .issuers.letsencrypt.staging.solvers[].http01.ingress.ingressClassName
```
Comment (anders-elastisys) [inline migration/v0.37/prepare/60-move-issuer-ingress-class.sh:11]: Could there be [`extraIssuers`](https://github.com/elastisys/compliantkubernetes-apps/blob/simonl/upgrade-cert-manager/config/config/common-config.yaml#L687) that should be be changed to use `ingressClassName` if they are using the `class` field as well?

**Review by anders-elastisys — APPROVED** [review]
LGTM, maybe using the `ingressClassName` instead can be a dev notice as well if we want to suggest application developers to use it instead?

**Comment by lunkan93** [comment]
Added an Application Developer notice

File: migration/v0.37/prepare/60-move-issuer-ingress-class.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+yq_move common .issuers.extraIssuers[].spec.acme.solvers[].http01.ingress.class .issuers.extraIssuers[].spec.acme.solvers[].http01.ingress.ingressClassName
```
Comment (aarnq) [inline migration/v0.37/prepare/60-move-issuer-ingress-class.sh:12]: Have you tested these when there are multiple set? I don't think the `yq_move` really works within arrays.

File: migration/v0.37/prepare/60-move-issuer-ingress-class.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+yq_move sc .issuers.letsencrypt.prod.solvers[].http01.ingress.class .issuers.letsencrypt.prod.solvers[].http01.ingress.ingressClassName
+yq_move sc .issuers.letsencrypt.staging.solvers[].http01.ingress.class .issuers.letsencrypt.staging.solvers[].http01.ingress.ingressClassName
+yq_move sc .issuers.extraIssuers[].spec.acme.solvers[].http01.ingress.class .issuers.extraIssuers[].spec.acme.solvers[].http01.ingress.ingressClassName
+fi
+
+if [[ "${CK8S_CLUSTER}" =~ ^(wc|both)$ ]]; then
+yq_move wc .issuers.letsencrypt.prod.solvers[].http01.ingress.class .issuers.letsencrypt.prod.solvers[].http01.ingress.ingressClassName
+yq_move wc .issuers.letsencrypt.staging.solvers[].http01.ingress.class .issuers.letsencrypt.staging.solvers[].http01.ingress.ingressClassName
+yq_move wc .issuers.extraIssuers[].spec.acme.solvers[].http01.ingress.class .issuers.extraIssuers[].spec.acme.solvers[].http01.ingress.ingressClassName
+fi
```
Comment (aarnq) [inline migration/v0.37/prepare/60-move-issuer-ingress-class.sh:24]: Indent inside if statement.

File: migration/v0.37/prepare/60-move-issuer-ingress-class.sh (hunk 2 above)
Comment (lunkan93) [reply migration/v0.37/prepare/60-move-issuer-ingress-class.sh:12]: Yeah I wasn't sure if it was going to work so I tested it and found that it actually did :+1:

File: migration/v0.37/prepare/60-move-issuer-ingress-class.sh (hunk 3 above)
Comment (lunkan93) [reply migration/v0.37/prepare/60-move-issuer-ingress-class.sh:24]: Working on fixing some other things in the migration script too :+1:

**Review by crssnd — APPROVED** [review]
didn't test, but look good

**Pull request merged**